src/flint/flint_base/flint_base.pyx

```diff
@@ -66,14 +66,50 @@ cdef class flint_poly(flint_elem):
 
     def roots(self):
         """
-        Deprecated function.
-        
-        To recover roots of a polynomial, first convert to acb:
-
-        acb_poly(input_poly).roots()
+        Computes all the roots in the base ring of the polynomial.
+        Returns a list of all pairs (*v*, *m*) where *v* is the 
+        integer root and *m* is the multiplicity of the root
+
+            >>> from flint import fmpz_poly
+            >>> fmpz_poly([]).roots()
+            []
+            >>> fmpz_poly([1]).roots()
+            []
+            >>> fmpz_poly([2, 1]).roots()
+            [(-2, 1)]
+            >>> fmpz_poly([1, 2]).roots()
+            []
+            >>> fmpz_poly([12, 7, 1]).roots()
+            [(-3, 1), (-4, 1)]
+            >>> (fmpz_poly([1, 2]) *  fmpz_poly([-3,1]) *  fmpz_poly([1, 2, 3]) *  fmpz_poly([12, 7, 1])).roots()
+            [(-3, 1), (-4, 1), (3, 1)]
+            >>> from flint import nmod_poly
+            >>> nmod_poly([1], 11).roots()
+            []
+            >>> nmod_poly([1, 2, 3], 11).roots()
+            [(8, 1), (6, 1)]
+            >>> nmod_poly([1, 6, 1, 8], 11).roots()
+            [(5, 3)]
+            >>> from flint import fmpq_poly
+            >>> fmpq_poly([1,2]).roots()
+            [(-1/2, 1)]
+            >>> fmpq_poly([2,1]).roots()
+            [(-2, 1)]
+            >>> f = fmpq_poly([fmpq(1,3), fmpq(3,5)]) * fmpq_poly([fmpq(4,11), fmpq(9)])
+            >>> f.roots()
+            [(-4/99, 1), (-5/9, 1)]
```

Comment:
It would be better to add most of these examples just to the tests rather than the docstring. The primary purpose of the docstring should be for users to read as documentation rather than to function as a test suite. In that sense it is probably also good to mention how to find complex roots here because I expect some users would come to this function expecting it to do that.